compiler/iface/MkIface.hs

```diff
@@ -713,6 +713,8 @@ addFingerprints hsc_env mb_old_fingerprint iface0 new_decls
     (non_orph_fis,   orph_fis)   = mkOrphMap ifFamInstOrph (mi_fam_insts iface0)
     fix_fn = mi_fix_fn iface0
     ann_fn = mkIfaceAnnCache (mi_anns iface0)
+    -- Do not allow filenames to affect the interface.
+    usages = [ case u of UsageFile _ fp -> UsageFile "" fp; _ -> u | u <- mi_usages iface0 ]
```

Comment:
I find this hard to read. And I really got confused with the naming here. Especially as I read `fp` as file path, and not fingerprint.

How about
```hs
-- anonUsage drops the file path and retains the fingerprint from the UsageFile,
-- so that the path does not affect the interface.
anonUsage :: Usage -> Usage
anonUsage (UsageFile _path hash) = UsageFile "" hash
anonUsage u = u
```
and using `map anonUsage (mi_usages iface0)`?

Reply:
It's a forward port of the original diff and I don't have a strong style preference in this case. If you send me a pull request for the branch in my fork, I'll merge it and it will show up here. Would that work for you?